README.md

```diff
@@ -26,36 +27,86 @@ available as
-    -   [Grafana Configuration](#grafana-configuration)
-    -   [Setting up Subscriptions](#setting-up-subscriptions)
-        -   [Aggregate Motion Sensor Count Events](#aggregate-motion-sensor-count-events)
-        -   [Sample Lamp Luminosity](#sample-lamp-luminosity)
-    -   [Time Series Data Queries (CrateDB)](#time-series-data-queries-cratedb)
-        -   [Read Schemas](#read-schemas)
-        -   [Read Tables](#read-tables)
-        -   [List the first N Sampled Values](#list-the-first-n-sampled-values)
-        -   [List N Sampled Values at an Offset](#list-n-sampled-values-at-an-offset)
-        -   [List the latest N Sampled Values](#list-the-latest-n-sampled-values)
-        -   [List the Sum of values over a time period](#list-the-sum-of-values-over-a-time-period)
-        -   [List the Minimum Values over a Time Period](#list-the-minimum-values-over-a-time-period)
-        -   [List the Maximum Values over a Time Period](#list-the-maximum-values-over-a-time-period)
-        -   [List the Average Values over a Time Period](#list-the-average-values-over-a-time-period)
--   [Accessing Time Series Data Programmatically](#accessing-time-series-data-programmatically)
-    -   [Displaying CrateDB data as a Grafana Dashboard](#displaying-cratedb-data-as-a-grafana-dashboard)
-        -   [Logging in](#logging-in)
-        -   [Configuring a Data Source](#configuring-a-data-source)
-        -   [Configuring a Dashboard](#configuring-a-dashboard)
--   [Next Steps](#next-steps)
+- [Contents](#contents)
+- [Persisting and Querying Time Series Data (CrateDB)](#persisting-and-querying-time-series-data-cratedb)
+    - [Analyzing time series data](#analyzing-time-series-data)
+            - [Grafana](#grafana)
+            - [Device Monitor](#device-monitor)
+            - [Device History](#device-history)
+- [Architecture](#architecture)
+- [Prerequisites](#prerequisites)
+    - [Docker and Docker Compose](#docker-and-docker-compose)
+    - [Cygwin for Windows](#cygwin-for-windows)
+- [Start Up](#start-up)
+- [Connecting FIWARE to a CrateDB Database via QuantumLeap](#connecting-fiware-to-a-cratedb-database-via-quantumleap)
+    - [CrateDB Database Server Configuration](#cratedb-database-server-configuration)
+    - [QuantumLeap Configuration](#quantumleap-configuration)
+    - [Grafana Configuration](#grafana-configuration)
+        - [Generating Context Data](#generating-context-data)
+    - [Setting up Subscriptions](#setting-up-subscriptions)
+        - [Aggregate Motion Sensor Count Events](#aggregate-motion-sensor-count-events)
+            - [:one: Request:](#one-request)
+        - [Sample Lamp Luminosity](#sample-lamp-luminosity)
+            - [:two: Request:](#two-request)
+    - [Time Series Data Queries (CrateDB)](#time-series-data-queries-cratedb)
+        - [:three: Checking Data persistence](#three-checking-data-persistence)
+            - [Using QuantumLeap API](#using-quantumleap-api)
```

Comment:
Rather than switching between CrateDB and QuantumLeap API, please move CrateDB requests down to the end of the tutorial